pkg/oc/clusteradd/componentinstall/create_secret.go

```diff
@@ -0,0 +1,116 @@
+package componentinstall
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/openshift/origin/pkg/oc/clusterup/coreinstall/kubeapiserver"
+	"github.com/openshift/origin/pkg/oc/clusterup/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/oc/clusterup/docker/run"
+	"github.com/openshift/origin/pkg/oc/errors"
+)
+
+type DockerConfigSecret struct {
```

Comment:
I don't see us getting super strict on what components can be added, but are you sure this one is worth the effort?  Seems fairly narrow and easy to add after the fact

Reply:
what do you propose instead?

Reply:
since i've moved this per your other comment, not sure if this comment is still applicable or not